### PR TITLE
Example accessing enum companion

### PR DESCRIPTION
**Pull request by som-snytt**
Fixes https://github.com/lampepfl/dotty/issues/11249

### CODE REVIEW

**Comment by som-snytt** [comment]
For "future reference", so to speak.

```
object ExoPlanetaryData:
  final val VulcanMass   = 5.331e+24
  final val VulcanRadius = 6.15297e6
import ExoPlanetaryData.*

import Planet.{VenusMass as VM, *}
enum Planet(mass: Double, radius: Double):
  private final val (MercuryMass @ _, MercuryRadius @ _) = (3.303e+23, 2.4397e6)

  case Mercury extends Planet(MercuryMass, MercuryRadius)             // Not found
  case Venus   extends Planet(VM, VenusRadius)                        // illegal reference
  case Earth   extends Planet(Planet.EarthMass, Planet.EarthRadius)   // ok
  case Vulcan  extends Planet(VulcanMass, VulcanRadius)               // ok
object Planet:
  private final val (VenusMass @ _, VenusRadius @ _) = (4.869e+24, 6.0518e6)
  private final val (EarthMass @ _, EarthRadius @ _) = (5.976e+24, 6.37814e6)
end Planet
```

**Comment by bishabosha** [comment]
@som-snytt would you be able to check the commit I pushed please?

**Comment by som-snytt** [comment]
@bishabosha I see you replaced royal "we"? I assumed you were required to use "royal we" as a condition of citizenship. But I am not bound to convention.

**Comment by bishabosha** [comment]
> I see you replaced royal "we"? I assumed you were required to use "royal we" as a condition of citizenship. But I am not bound to convention.

I've noticed I have actually already used "we" in some other parts of this document (like for deprecation of enum cases) 😂

**Comment by som-snytt** [comment]
I will leave it to your discretion.

I recently joked that so-and-so should just change their pronouns to "we". I've forgotten the context, as ought to do when there is no implicit context.

File: docs/_docs/reference/enums/enums.md [hunk 1]
```diff
@@ -89,6 +90,33 @@ object Planet:
 end Planet
 ```
 
+### Restrictions on Enum Cases
+
+Enum case declarations are similar to secondary constructors:
+they are scoped outside of the enum template, despite being declared within it.
+This means that enum case declarations can not access inner members of the
```
Comment (som-snytt) [inline docs/_docs/reference/enums/enums.md:97]: missed opportunity `cannot`, an omission I've made since childhood.

Comment (bishabosha) [reply docs/_docs/reference/enums/enums.md]: thanks!

**Comment by oscar-broman** [comment]
Is it really necessary to use that pattern matching syntax? The example would look a bit simpler if the names started with lowercase letters.

I'm just thinking it might confuse novices reading the documentation, especially since enums is a simple concept that people will want to learn initially as they learn a new language.

**Comment by som-snytt** [comment]
@oscar-broman I don't mind following up if you suggest a delta. I'm not sure what you mean, as it has been a few months.

I'm not sure the text is intended for novices? I mean, I can barely follow it.

**Comment by oscar-broman** [comment]
I just meant the example could look like this instead:
```scala
  import Planet.*
  enum Planet(mass: Double, radius: Double):
    private final val (mercuryMass, mercuryRadius) = (3.303e+23, 2.4397e6)

    case Mercury extends Planet(mercuryMass, mercuryRadius)             // Not found
    case Venus   extends Planet(venusMass, venusRadius)                 // illegal reference
    case Earth   extends Planet(Planet.earthMass, Planet.earthRadius)   // ok
  object Planet:
    private final val (venusMass, venusRadius) = (4.869e+24, 6.0518e6)
    private final val (earthMass, earthRadius) = (5.976e+24, 6.37814e6)
  end Planet
```

**Comment by bishabosha** [comment]
I'll open a PR

**Comment by som-snytt** [comment]
@oscar-broman Thanks for clarifying. The syntax was "pedagogical" because normally we expect "constants" to be uppercase. The idiom shown is how to achieve that.

**Comment by som-snytt** [comment]
@bishabosha I'm agnostic as to whether the example is done the right way or the wrong way. :)

**Comment by bishabosha** [comment]
see https://github.com/lampepfl/dotty/pull/14556

**Pull request merged**